stuffListApi/public/controllers/controller.js

```diff
@@ -0,0 +1,176 @@
+var app = angular.module('myStuffApp', []);
+
+app.factory('Stuffs', ['$http', function($http) {
+    var stuffsFactory = {};
+
+    stuffsFactory.getStuffs = function() {
+        return $http.get('/stuffs');
+    }
+
+    stuffsFactory.createStuff = function(body) {
+        return $http.post('/stuffs', body);
+    }
+
+    stuffsFactory.readStuff = function(stuffId) {
+        return $http.get('/stuffs/' + stuffId);
+    }
+
+    stuffsFactory.updateStuff = function(stuffId, body) {
+        return $http.put('/stuffs/' + stuffId, body);
+    }
+
+    stuffsFactory.deleteStuff = function(stuffId) {
+        return $http.delete('/stuffs/' + stuffId);
+    }
+
+    stuffsFactory.deleteAllStuff = function() {
+        return $http.delete('/stuffs');
+    }
+
+    return stuffsFactory;
+}]);
+
+app.controller('StuffController', ['$scope', '$filter', 'Stuffs', function($scope, $filter, Stuffs) {
+    $scope.currentStuffId = null;
+
+    var showSpinner = function(displaySpinner) {
+        if (displaySpinner) {
+            $('#spinner').show();
+        } else {
+            $('#spinner').hide();
+        }
+    };
+
+    var refresh = function() {
+        Stuffs.getStuffs().then(function successCallback(response) {
+            $scope.stuffs = response.data;
+            putRecordInCategory();
+            showSpinner(false);
+        }, function errorCallback(response) {
+            console.log(response.data);
+            showSpinner(false);
+        });
+    };
+
+    var readRecord = function(stuffId, modalTitle, action) {
+        var record = null;
+        Stuffs.readStuff(stuffId).then(function successCallback(response) {
+            record = response.data;
+            $scope.stuffName = record.name;
+            $scope.stuffCategory = record.category;
+            $scope.currentStuffId = record._id;
+            genericModalConfig(modalTitle, action);
+        }, function errorCallback(response) {
+            console.log(response.data);
+        });
+    };
+
+    var putRecordInCategory = function() {
+        var categories = [];
+        $scope.stuffCategorized = categories;
+
+        for (var i = 0; i < $scope.stuffs.length; i++) {
```

Comment:
Some comments particularly here would aid to know what the intent for the logic is prior to reviewing/maintaining the code. This way if for example, I was trying to fix a bug, I would have an idea of what to expect from the following lines and where my fix could be placed at. (Think about your future self also, he'll be looking into the code you write today).